app/controllers/shops_controller.rb

```diff
@@ -0,0 +1,18 @@
+class ShopsController < ApplicationController
+  before_action :set_shop, only: :show
+
+  def show
+    @avg_rating_by_month = @shop.products
+                                .get_average_rating(time_range)
+  end
+
+  private
+
+  def set_shop
+    @shop = Shop.find_by(id: params[:id]) or not_found
```

Comment:
FYI: we can simply do like this and Rails will respond with 404 status page if the shop is not found:
```suggestion
    @shop = Shop.find(params[:id])
```